### PR TITLE
Align MO pip requirements

### PR DESCRIPTION
**Pull request by ilya-naumov**
### Details:
 - Set MO's pip dependency requirements to ~= and remove TF1

### Tickets:
 - internal 48358

### CODE REVIEW

File: model-optimizer/requirements_tf.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
-tensorflow>=1.15.2,<2.0; python_version < "3.8"
-tensorflow>=2.2; python_version >= "3.8"
-networkx>=1.11
-numpy>=1.14.0,<1.19.0
+tensorflow>=1.15.5,<2.0
+networkx~=2.5
+numpy~=1.19.5
```
Comment (lazarevevgeny) [inline model-optimizer/requirements_tf.txt:3]: TF 1.X requires numpy<1.19.0.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/requirements.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-protobuf>=3.6.1
-onnx>=1.1.2
+tensorflow>=2.4.1
+mxnet>=1.2.0 ; sys_platform == 'win32'
+mxnet>=1.7.0.post2 ; sys_platform == 'linux'
+networkx~=2.5
+numpy~=1.19.5
+protobuf>=3.15.6
+onnx>=1.8.1
 test-generator==0.1.1
```
Comment (lazarevevgeny) [inline model-optimizer/requirements.txt:8]: Please, remove `test-generator=0.1.1` from all files except requirements_dev.txt. This change has been merged in the MO.

**Comment by lazarevevgeny** [comment]
@achetver , please, take a look at this PR. Does MO version checker works correctly for the updated requirements?

**Review by lazarevevgeny — APPROVED** [review]

**Comment by achetver** [comment]
@lazarevevgeny MO version checker in [PR](https://github.com/openvinotoolkit/openvino/pull/5342) works correctly with `~=` signs but I don't see support of `sys_platform == '...'` environment marker and during environment check these markers will be ignored.

**Comment by generalova-kate** [comment]
@achetver could you add support sys_platform on this week?

**Comment by achetver** [comment]
@generalova-kate Yes, I can. But it is necessary to use different versions of mxnet depending on platform?

**Comment by ilya-naumov** [comment]
@achetver  Windows package of `mxnet 1.7.0.post2` conflicts with `numpy` and `requests` (Windows wheels isn't available for newer versions on PyPi). We use `mxnet 1.2.0` on Windows because it doesn't cause dependency conflicts on this environment and we can't use same version on Linux because it requires old `numpy`. You can find more info at the ticket.

File: model-optimizer/requirements_mxnet.txt [hunk 3]
```diff
@@ -1,6 +1,7 @@
-mxnet>=1.0.0,<=1.7.0
-networkx>=1.11
-numpy>=1.14.0
-defusedxml>=0.5.0
-urllib3>=1.25.9
-requests>=2.20.0
+mxnet~=1.2.0 ; sys_platform == 'win32'
```
Comment (achetver) [inline model-optimizer/requirements_mxnet.txt:1]: Are whitespaces between module version and `;` necessary? If not, please remove them. Note as we use environment markers `tensorflow>=1.15.2,<2.0; python_version < "3.8"`

**Comment by lazarevevgeny** [comment]
PR https://github.com/openvinotoolkit/openvino/pull/5437 has been merged. Please, merge the recent master to your branch and re-validate again.

**Pull request merged**